### DISCUSSION TITLE
mixcr with 10x Visium data

### DISCUSSION THREAD

**Discussion by lymphomatic** [milaboratory/mixcr#1718] [Q&A] [answered]
I am trying to extract TCR and BCR clonotypes from 10x Visium data (standard library without VDJ enrichment), but I am getting no results using the standard 10x-sc-5gex preset (no alignment), even with addition of `--tag-pattern "^(CELL:N{16})(UMI:N{12})\^(R2:*)` correction.
```
mixcr analyze 10x-sc-5gex \
--species hsa \
--tag-pattern "^(CELL:N{16})(UMI:N{12})\^(R2:*)"  \
"RS23_10390_10B_S6_R1_001.fastq.gz"  "RS23_10390_10B_S6_R2_001.fastq.gz" \
result
```
I have tried to use the BAM file from the 10x SpaceRanger output, but I get consistently the error (I have tried with several different BAM files from different samples):
```
CommandLine$ExecutionException: Error while running command align htsjdk.samtools.SAMFormatException: SAM validation error: ERROR::INVALID_CIGAR:Record 155943416, Read name VH00708:358:22232VMNX:2:2502:32572:54393, CIGAR should have > zero elements for mapped read.
```
I have also tried using the simple `rna-seq` preset, and using the alternative `--tag-pattern "^(UMI:N{28})\^(R2:*)` but without difference.
Am I doing something patently wrong here? Or, alternatively, is it possible there might be a preset for (possibly pseudo-bulked) 10x Visium-specific preset?

Thanks for any insight!
Adam

**Comment by mizraelson** [comment]
Hi, sorry for the late reply. Generally speaking it is unlikely to recover decent immune repertoire from non targeted 10x data especially with 3' chemistry (default for 10x Visium).

Nevertheless, I would recommend to try the following:

```
mixcr analyze generic-ht-single-cell-fragmented-with-umi \
​     --rna \
     --species hsa \
     --tag-pattern "^(CELL:N{16})(UMI:N{12})\^(R2:*)" \
     -Massemble.consensusAssemblerParameters.assembler.maxConsensuses=10 \
     sample_R1.fastq.gz \
     sample_R2.fastq.gz \
     sample_result
```

Since theoretically there can be multiple cells marked with the same spatial barcode I adjust the maxConsensuses number to 10 per chain (TRA, TRB, IGH, IGL, IGK...) (default is 2).

**Reply by lymphomatic** [reply]
Thank you very much for the suggestion! Despite all attempts, every run ends up with the result:
`
Reads dropped due to the lack of a clone sequence, percent of total: 750 (0%)
`
I think I can assume that this is just the lack of adequate sequences in the data.
My only concern is that all alignments on review from the _exportAlignmentsPretty_ module looks very similar:
`                                                    <J                  CDR3><FR4              
             _  V  I  L  F  F  F  F  F  F  F  F  F  F  T  T  G  W  F  K  I  F  A  E  G  T  K    
  Quality    48888488488888888888888888888448884888848888888888888888888888888888888888888888   
  Target0  0 CCGTCATTCTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTACCACTGGTTGGTTCAAGATATTTGCTGAAGGGACTAAG 79  Score
TRGJP1*00 21                                         taccactggttggttcaagatatttgctgaagggactaag 60  250

                          
              L  I  V _   
  Quality    8888888888   
  Target0 80 CTCATAGTAA 89  Score
TRGJP1*00 61 ctcatagtaa 70  250

>>> Read ids: 2520780`
- which makes me worried that perhaps I am missing some parameter that prevents proper assembly.

**Reply by mizraelson** [reply]
Can you share the alignment, refine and assemble txt report files?

**Reply by lymphomatic** [reply]
Of course, thank you for taking a look at them! 
It seems that all reads are dropped "due to the lack of a clone sequence", so perhaps they are just too short.

[res.align.report.txt](https://github.com/user-attachments/files/16570399/res.align.report.txt)
[res.assemble.report.txt](https://github.com/user-attachments/files/16570400/res.assemble.report.txt)
[res.refine.report.txt](https://github.com/user-attachments/files/16570401/res.refine.report.txt)

**Reply by mizraelson** [reply, marked as answer]
Given that it's a 3' non-enriched library, we have relatively few reads to begin with:
`Successfully aligned reads: 260,361 (0.14%)`

89% of these reads cover the region outside the CDR3:
`No CDR3 parts alignments, percent of successfully aligned: 231,687 (88.99%)`

And 11% partially cover the CDR3 region:
`Partially aligned reads, percent of successfully aligned: 28,674 (11.01%)`

MiXCR has a partial assembly step that rescues some CDR3s if two pairs of reads combined cover the same CDR3 completely, but that doesn't seem to be the case here. Even after this step, none of the alignments fully cover any complete CDR3 region sequence:
`Reads dropped due to the lack of a clone sequence, percent of total: 250,869 (100%)`
